### PR TITLE
[Bug][Connectors] Text And Json WriteStrategy lost the sinkColumnsIndexInRow

### PR DESCRIPTION
**Pull request by lvshaokang**
## Purpose of this pull request

refer #3815

Fix text and json write strategy lost the sinkColumnsIndexInRow

## Check list

* [ ] Code changed are covered with tests, or it does not need tests for reason:
* [ ] If any new Jar binary package adding in your PR, please add License Notice according
  [New License Guide](https://github.com/apache/incubator-seatunnel/blob/dev/docs/en/contribution/new-license.md)
* [ ] If necessary, please update the documentation to describe the new feature. https://github.com/apache/incubator-seatunnel/tree/dev/docs
* [ ] If you are contributing the connector code, please check that the following files are updated:
  1. Update change log that in connector document. For more details you can refer to [connector-v2](https://github.com/apache/incubator-seatunnel/tree/dev/docs/en/connector-v2)
  2. Update [plugin-mapping.properties](https://github.com/apache/incubator-seatunnel/blob/dev/plugin-mapping.properties) and add new connector information in it
  3. Update the pom file of [seatunnel-dist](https://github.com/apache/incubator-seatunnel/blob/dev/seatunnel-dist/pom.xml)

### CODE REVIEW

**Review by TaoZex — DISMISSED** [review]
+1

File: seatunnel-api/src/main/java/org/apache/seatunnel/api/table/type/SeaTunnelRow.java [hunk 1]
```diff
@@ -95,6 +96,12 @@ public SeaTunnelRow copy(int[] indexMapping) {
         return newRow;
     }
 
+    public SeaTunnelRow copy(List<Integer> indexMapping) {
+        return copy(indexMapping.stream()
+            .mapToInt(Integer::intValue)
+            .toArray());
+    }
+
```
Comment (hailin0) [inline seatunnel-api/src/main/java/org/apache/seatunnel/api/table/type/SeaTunnelRow.java]: remove this method, move to caller location

**Comment by hailin0** [comment]
@TyrantLucifer

File: seatunnel-api/src/main/java/org/apache/seatunnel/api/table/type/SeaTunnelRow.java (hunk 1 above)
Comment (TyrantLucifer) [reply seatunnel-api/src/main/java/org/apache/seatunnel/api/table/type/SeaTunnelRow.java]: Aggre with @hailin0

Comment (lvshaokang) [reply seatunnel-api/src/main/java/org/apache/seatunnel/api/table/type/SeaTunnelRow.java]: @hailin0 @TyrantLucifer done. thanks for you review

**Review by TaoZex — APPROVED** [review]

**Comment by EricJoy2048** [comment]
@TyrantLucifer  PTAL

**Review by TyrantLucifer — APPROVED** [review]

**Review by EricJoy2048 — APPROVED** [review]
+1

**Pull request merged**